### ISSUE TITLE
[BUG] bot/user name shenanigans

### ISSUE THREAD

**Issue by StefanDanielSchwarz** [SillyTavern/SillyTavern#648]
**Have you searched for similar [bugs](https://github.com/SillyTavern/SillyTavern/issues?q=)?**
Yes

**Describe the bug**
I've noticed multiple cases of the user's name changing to "You" and the bot's name to "Bot" unexpectedly, ruining generations. This seems to happen whenever an auto-generation is invoked (instead of a normal generation started by sending a user message normally), be it through a command, summarization, or group chat's self responses.

**To Reproduce**
1. Chat normally and check the logs, {{user}} and {{char}} get replaced with user's and character's name
2. Start an auto-generation, e. g. by using the command `/continue`, `/bgauto`, or any other way to make the AI respond on its own
3. Check the logs, {{user}} is now "You" and {{char}} is "Bot", despite the character and user persona having other names

**Expected behavior**
{{user}} and {{char}} are always the proper user and character names and don't unintentionally revert to "You" and "Bot".

**Screenshots**
–

**Logs**
Input: {"n": 1, "rep_pen": 1.18, "temperature": 0, "top_p": 0, "top_k": 1, "top_a": 0, "typical": 1, "tfs": 1, "rep_pen_range": 2048, "rep_pen_slope": 0, "sampler_order": [0, 1, 2, 3, 4, 5, 6], "quiet": false, "max_context_length": 2048, "max_length": 250, "prompt": "## Bot\n- You're \"Bot\" in this never-ending roleplay with \"You\".\n…\n[Continue the following message. Do not include ANY parts of the original message. Use capitalization and punctuation as if your reply is a part of the original message:\n…", "stop_sequence": ["</s>", "\n##", "\nYou:", "\n# ", "##", "\nInstruction:", "\nUser:"]}

**Desktop (please complete the following information):**
 - OS/Device: Windows 11
 - Environment: local
 - Node.js version (if applicable): v18.15.0
 - Browser: Firefox
 - Generation API: koboldcpp
 - Branch: main (1.8.2)
 - Model: LLaMA 33B

**Additional context**
I created a user persona, but while it is selected, it's not locked as far as I can tell.

**Comment by StefanDanielSchwarz** [comment]
Sorry, you're right, I should have mentioned it. And I got confused, posting the koboldcpp log instead of the SillyTavern one.

I've continued to investigate and think the proxy problem is caused by there being two system messages, the first being the jailbreak (which the proxy evaluates), and the second one including the auto-gen prompt like '[Continue the following message. …"). So when there are two system messages, the proxy gets confused.

I'll bring it up in the proxy's bug tracker instead. Sorry to bother you, next time I'll be more attentive.
